src/backend/routers/auth.py

```diff
@@ -317,3 +317,74 @@ def log_and_redirect_err(error_message: str):
     response = RedirectResponse(redirect_uri)
 
     return response
+
+
+@router.delete("/tool/auth")
+async def delete_tool_auth(
+    request: Request, session: DBSessionDep, ctx: Context = Depends(get_context)
+):
+    """
+    Endpoint to delete Tool Authentication.
+
+    If completed, the corresponding ToolAuth for the requesting user is removed from the DB.
+
+    Args:
+        request (Request): current Request object. Note that the request body should contain the tool_id parameter as the tool to delete.
+        session (DBSessionDep): Database session.
+        ctx (Context): Context object.
+
+    Returns:
+        {"status": "success"} on successful deletion.
+
+    Raises:
+        HTTPException: If there was an error deleting the tool auth.
+    """
+
+    logger = ctx.get_logger()
+
+    def log_and_return_error(error_message: str):
+        logger.error(event=error_message)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"{error_message}",
+        )
+
+    user_id = ctx.get_user_id()
+
+    try:
+        request_body = await request.json()
+        tool_id = request_body["tool_id"]
+    except Exception as e:
+        log_and_return_error("Error parsing tool_id from request body.")
+
+    if user_id is None or user_id == "" or user_id == "default":
+        log_and_return_error("User ID not found.")
+
+    if not isinstance(tool_id, str):
+        log_and_return_error(
+            "tool_id must be present in the request body and must be a string."
+        )
+
+    if tool_id != ToolName.Google_Drive:
```

Comment:
Does it need to be tool specific? Couldn't we run 
`tool_auth_crud.delete_tool_auth(session, user_id, self.TOOL_ID)`
here?

Reply:
After the google drive tool/sync service session today, I thought it would make the most sense to put the tool auth deletion function in the `BaseToolAuthentication` class. Please let me know if there is a better place for it!